Adding GEN fragment (GluGluTo2Jets_ExHume) specific for PPS wf testing

mundim opened the pull request:
#### PR description:

Add a GEN fragment (ExHume generator, gluongluon -> 2 Jets, 14 TeV, central mass between 300 and 2000 GeV) to produce protons inside PPS acceptance to allow it to be properly tested.

#### PR validation:
scram b runtests
scram b code-checks
scram b code-format

ran without any issues.

standard runTheMatrix --jobsReports -l limited --ibeos
produced no error

runTheMatrix.py --what upgrade -l 11725.0,11925.0,11925.98,11925.99,12125.0,12325.0,12325.98,12325.99 -t 4 --command='-n 5

where the wf are the ones in which GluGluTo2Jets is present (as advised in #32352), produced no error for the ones with NO PU, but it was reported missing files for the PU ones

mundim committed:
Adding GEN fragment (GluGluTo2Jets_ExHume) specific for PPS wf testing

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32765/20923

- This PR adds an extra 36KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #32128, #32738, #27983, #31854, #32576

cmsbuild commented:
A new Pull Request was created by  @mundim  for master.

It involves the following packages:

Configuration/Generator
Configuration/PyReleaseValidation


@SiewYan, @mkirsano, @jordan-martins, @chayanit, @wajidalikhan, @kpedro88, @cmsbuild, @GurpreetSinghChahal, @srimanob, @agrohsje, @alberto-sanchez can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @makortel, @fabiocos, @slomeo this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

srimanob commented:
test parameters:
- workflows = 11725.0,11925.0,11925.98,11925.99,12125.0,12325.0,12325.98,12325.99

srimanob commented:
Please test

mundim commented:
the wf 12325.0, 11925.0, 12325.98, 11925.98, 12325.99 and 11925.99 (the ones with PU) failed in my area because of a missing secondary file. I hope it will not happen in the official testing area.

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dfda86/12591/summary.html
**COMMIT**: 4425c6768dabf98703c25d7584fb9d1711c6f1fe
**CMSSW**: CMSSW_11_3_X_2021-01-28-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32765/12591/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

mundim commented:
@chayanit sorry, I forgot to mention you here.

chayanit commented on Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:
```diff
@@ -1288,4 +1288,5 @@ def __init__(self, howMuch, dataset):
     ('TenTau_E_15_500_Eta3p1_pythia8_cfi', UpgradeFragment(Kby(9,100),'TenTau_15_500_Eta3p1')),
     ('QCD_Pt_1800_2400_14TeV_TuneCP5_cfi', UpgradeFragment(Kby(9,50), 'QCD_Pt_1800_2400_14')),
     ('DisplacedSUSY_stopToBottom_M_800_500mm_TuneCP5_14TeV_pythia8_cff', UpgradeFragment(Kby(9,50),'DisplacedSUSY_stopToB_M_800_500mm_14')),
```
Would you mind to help me also update this line so I don't need to open new PR?
`('DisplacedSUSY_stopToBottom_M_800_500mm_TuneCP5_14TeV_pythia8_cff', UpgradeFragment(Kby(9,50),'DisplacedSUSY_14TeV')),`

mundim replied:
Sure.

chayanit commented on Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:
```diff
@@ -1288,4 +1288,5 @@ def __init__(self, howMuch, dataset):
     ('TenTau_E_15_500_Eta3p1_pythia8_cfi', UpgradeFragment(Kby(9,100),'TenTau_15_500_Eta3p1')),
     ('QCD_Pt_1800_2400_14TeV_TuneCP5_cfi', UpgradeFragment(Kby(9,50), 'QCD_Pt_1800_2400_14')),
     ('DisplacedSUSY_stopToBottom_M_800_500mm_TuneCP5_14TeV_pythia8_cff', UpgradeFragment(Kby(9,50),'DisplacedSUSY_stopToB_M_800_500mm_14')),
+    ('GluGluTo2Jets_M_300_2000_14TeV_Exhume_cff',UpgradeFragment(Kby(9,100),'GluGluTo2Jets_M_300_2000_14TeV')),
```
I'm afraid the string is a bit too long. I would request to change to `('GluGluTo2Jets_M_300_2000_14TeV_Exhume_cff',UpgradeFragment(Kby(9,100),'GluGluTo2Jets_14TeV')),`

chayanit commented:
please test
@srimanob I don't think we can ask bot to test new workflows.

chayanit commented:
> @chayanit sorry, I forgot to mention you here.

no need. I'm included as L2 PdmV to assign this PR already. Could you commit changes as I requested? Thanks.

mundim committed:
Shortenning a string the wf configuration

mundim commented:
Done!

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32765/20928

- This PR adds an extra 20KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #32128, #32738, #27983, #31854, #32576

cmsbuild commented:
Pull request #32765 was updated. @SiewYan, @mkirsano, @jordan-martins, @chayanit, @wajidalikhan, @kpedro88, @cmsbuild, @GurpreetSinghChahal, @srimanob, @agrohsje, @alberto-sanchez can you please check and sign again.

srimanob commented:
> please test
> @srimanob I don't think we can ask bot to test new workflows.

Why?

chayanit commented:
> > please test
> > @srimanob I don't think we can ask bot to test new workflows.
> 
> Why?

I don't know the reason maybe @silviodonato or @qliphy can explain but usually new added wfs can't be tested by bot.

silviodonato commented:
@srimanob @chayanit yes, you can test the new workflows added with a PR. The problem here is that it is not possible to use `--what upgrade` in the PR test (cc @smuzaffar)

davidlange6 commented:
Perhaps these workflows should be in the default set for at least one geometry?

> On Jan 30, 2021, at 2:31 PM, Silvio Donato <notifications@github.com> wrote:
> 
> 
> @srimanob @chayanit yes, you can test the new workflows added with a PR. The problem here is that it is not possible to use --what upgrade in the PR test (cc @smuzaffar)
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

slava77 commented:
earlier on I pointed to where to add the wf number to have it in the general matrix https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_2017.py#L60
https://github.com/cms-sw/cmssw/issues/32356#issuecomment-768420402
I guess this was lost/forgotten.

agrohsje commented:
Would it make sense to update the SM parameters to more recent values? Why not using PDG2020? Just in case somebody is using it as a reference.

mundim commented:
> earlier on I pointed to where to add the wf number to have it in the general matrix https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_2017.py#L60
> [#32356 (comment)](https://github.com/cms-sw/cmssw/issues/32356#issuecomment-768420402)
> I guess this was lost/forgotten.

In fact, I have asked another question on top of your message, but I could not find it here... I asked to confirm the file name because PPS is not included yet in 2017 and I could not find an equivalent one for 2021.

slava77 commented:
> In fact, I have asked another question on top of your message, but I could not find it here... I asked to confirm the file name because PPS is not included yet in 2017 and I could not find an equivalent one for 2021.

2017 here is just a number; after looking at the file it should be obvious that it is used for most of Run2 and Run3.

chayanit commented:
Hello @mundim , could you update the SM parameters as requested by Alexander?

> Would it make sense to update the SM parameters to more recent values? Why not using PDG2020? Just in case somebody is using it as a reference.

And I think @slava77 requested to add your workflow to IB test. You just need to add 12325.0 (noPU) and 11925.0 (PU) to this line at the back https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_2017.py#L60 and also add clarification which sample they correspond to at this line https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_2017.py#L33 

Maybe @slava77 could confirm if we need both noPU and PU in IB test?

mundim committed:
Adding the GluGluTo2Jets (for PPS) wfs into relval_2017 config file

mundim commented:
> Hello @mundim , could you update the SM parameters as requested by Alexander?
> 
> > Would it make sense to update the SM parameters to more recent values? Why not using PDG2020? Just in case somebody is using it as a reference.
> 
> And I think @slava77 requested to add your workflow to IB test. You just need to add 12325.0 (noPU) and 11925.0 (PU) to this line at the back https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_2017.py#L60 and also add clarification which sample they correspond to at this line https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/relval_2017.py#L33
> 
> Maybe @slava77 could confirm if we need both noPU and PU in IB test?

Done. Thank you very much for your help, but just as a side comment, it is far from obvious to include wfs for a detector present only  in 2021 onwards in a file with 2017 in its name, this is not just a number. Anyhow, thanks for everyone again.
Cheers

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32765/21059

- This PR adds an extra 12KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_2017.py  modified in PR(s): #31854, #27983, #32128
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #31854, #27983, #32128, #32843, #32789

cmsbuild commented:
Pull request #32765 was updated. @SiewYan, @mkirsano, @jordan-martins, @chayanit, @wajidalikhan, @kpedro88, @cmsbuild, @GurpreetSinghChahal, @srimanob, @agrohsje, @alberto-sanchez can you please check and sign again.

chayanit commented on Configuration/PyReleaseValidation/python/relval_2017.py:
```diff
@@ -30,7 +30,7 @@
 #   2021 (DD4HEP: TTbar, ZMM)
 #        (ele guns 10, 35, 1000; pho guns 10, 35; mu guns 1, 10, 100, 1000, QCD 3TeV, QCD Flat)
-#        (ZMM, TTbar, ZEE, MinBias, TTbar PU, TTbar PU premix, ZEE PU, TTbar design)
+#        (ZMM, TTbar, ZEE, MinBias, TTbar PU, TTbar PU premix, ZEE PU,GluGluTo2Jets, TTbar design,GluGluTo2Jets)
```
sorry for being a bit picky but could you update to 
(ZMM, TTbar, ZEE, MinBias, TTbar PU, TTbar PU premix, ZEE PU, TTbar design, GluGluTo2Jets, GluGluTo2Jets PU) note for the space as well

mundim replied:
Don't be sorry, that's ok, thank you... I put that way because I imagined it should have a correspondence with the wf id, as in line 60, where I put them ordered by their number. Tell me if that is not the case. Thank you.

chayanit commented on Configuration/PyReleaseValidation/python/relval_2017.py:
```diff
@@ -57,7 +57,7 @@
            11624.911, 11642.911,
            11601.0,11602.0,11603.0,11604.0,11605.0,11606.0,11607.0,11608.0,11609.0,11630.0,11643.0,
-           11650.0,11634.0,11646.0,11640.0,11834.0,11834.99,11846.0,12024.0,
+           11650.0,11634.0,11646.0,11640.0,11834.0,11834.99,11846.0,11925.0,12024.0,12325.0
```
and also reorder here

mundim replied:
Ops.. ok, I'll change them right away.

chayanit commented:
@mundim have you updated the SM parameters? I think it should be updated since this is for Run3 as you mentioned. Perhaps @agrohsje can help pointing out which ones.

mundim committed:
Reordering the wf number and description in relval_2017.py

mundim commented:
> @mundim have you updated the SM parameters? I think it should be updated since this is for Run3 as you mentioned. Perhaps @agrohsje can help pointing out which ones.

I'm not sure I understood what you mean. Sorry.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32765/21071

- This PR adds an extra 12KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_2017.py  modified in PR(s): #31854, #27983, #32128
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #31854, #27983, #32128, #32843, #32789

cmsbuild commented:
Pull request #32765 was updated. @SiewYan, @mkirsano, @jordan-martins, @chayanit, @wajidalikhan, @kpedro88, @cmsbuild, @GurpreetSinghChahal, @srimanob, @agrohsje, @alberto-sanchez can you please check and sign again.

chayanit commented:
> > @mundim have you updated the SM parameters? I think it should be updated since this is for Run3 as you mentioned. Perhaps @agrohsje can help pointing out which ones.
> 
> I'm not sure I understood what you mean. Sorry.

Please double check SM parameters here https://github.com/cms-sw/cmssw/blob/db716a004f42d2c59cd8ece55d39dec8b8b359e4/Configuration/Generator/python/GluGluTo2Jets_M_300_2000_14TeV_Exhume_cff.py#L7-L23 I think at least the top mass is outdated :)

mundim commented:
> > > @mundim have you updated the SM parameters? I think it should be updated since this is for Run3 as you mentioned. Perhaps @agrohsje can help pointing out which ones.
> > 
> > 
> > I'm not sure I understood what you mean. Sorry.
> 
> Please double check SM parameters here
> 
> https://github.com/cms-sw/cmssw/blob/db716a004f42d2c59cd8ece55d39dec8b8b359e4/Configuration/Generator/python/GluGluTo2Jets_M_300_2000_14TeV_Exhume_cff.py#L7-L23
> 
> I think at least the top mass is outdated :)

Oh... Of course.. I'll check. Sorry for the delay in replying... I'm quite overloaded here..
Thanks.

mundim committed:
Updating/correcting SM value in the glugluTo2jet gen fragment

mundim commented:
> > > @mundim have you updated the SM parameters? I think it should be updated since this is for Run3 as you mentioned. Perhaps @agrohsje can help pointing out which ones.
> > 
> > 
> > I'm not sure I understood what you mean. Sorry.
> 
> Please double check SM parameters here
> 
> https://github.com/cms-sw/cmssw/blob/db716a004f42d2c59cd8ece55d39dec8b8b359e4/Configuration/Generator/python/GluGluTo2Jets_M_300_2000_14TeV_Exhume_cff.py#L7-L23
> 
> I think at least the top mass is outdated :)

You were right, some of them were outdated and there was one typo. Sorry about that and thank you very much to have called my attention.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32765/21101

- This PR adds an extra 40KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_2017.py  modified in PR(s): #31854, #32857, #27983, #32128
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #31854, #32857, #27983, #32128, #32868, #32843

cmsbuild commented:
Pull request #32765 was updated. @SiewYan, @mkirsano, @jordan-martins, @chayanit, @wajidalikhan, @kpedro88, @cmsbuild, @GurpreetSinghChahal, @srimanob, @agrohsje, @alberto-sanchez can you please check and sign again.

chayanit commented on Configuration/PyReleaseValidation/python/relval_2017.py:
```diff
@@ -57,7 +57,7 @@
            11624.911, 11642.911,
            11601.0,11602.0,11603.0,11604.0,11605.0,11606.0,11607.0,11608.0,11609.0,11630.0,11643.0,
-           11650.0,11634.0,11646.0,11640.0,11834.0,11834.99,11846.0,12024.0,
+           11650.0,11634.0,11646.0,11640.0,11834.0,11834.99,11846.0,12024.0,11925.0,12325.0
```
I think a comma (,) is missing after 12325.0

mundim replied:
Sorry again.

mundim committed:
correcting a missing comma in the list of wf ids

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32765/21113

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_2017.py  modified in PR(s): #27983, #32128, #31854
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #27983, #32128, #32868, #32843, #31854

cmsbuild commented:
Pull request #32765 was updated. @SiewYan, @mkirsano, @jordan-martins, @chayanit, @wajidalikhan, @kpedro88, @cmsbuild, @GurpreetSinghChahal, @srimanob, @agrohsje, @alberto-sanchez can you please check and sign again.

chayanit commented:
test parameters:

- workflows = 11925.0,12325.0

chayanit commented:
please test

cmsbuild commented:
-1

**Failed Tests**:   Build  Python3 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dfda86/12853/summary.html
**COMMIT**: 15510e4309da9983a54b01b6ba89915c108cbbd9
**CMSSW**: CMSSW_11_3_X_2021-02-11-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32765/12853/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation warning when building:  See details on the summary page.

#Python3

I found errors: 

 <pre>*** Error compiling 'src/Configuration/Generator/python/GluGluTo2Jets_M_300_2000_14TeV_Exhume_cff.py'...
</pre>

mundim committed:
correcting a misplaced comma

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32765/21120

- This PR adds an extra 40KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_2017.py  modified in PR(s): #27983, #32128, #31854
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #32886, #27983, #32128, #32868, #32843, #31854

cmsbuild commented:
Pull request #32765 was updated. @SiewYan, @mkirsano, @jordan-martins, @chayanit, @wajidalikhan, @kpedro88, @cmsbuild, @GurpreetSinghChahal, @srimanob, @agrohsje, @alberto-sanchez can you please check and sign again.

chayanit commented:
test parameters:

- workflows = 11925.0,12325.0

chayanit commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dfda86/12856/summary.html
**COMMIT**: be40ede9d5b80b6f142aeb6ef5b83b82d11d5118
**CMSSW**: CMSSW_11_3_X_2021-02-12-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32765/12856/install.sh` to create a dev area with all the needed externals and cmssw changes.


* **DAS Queries**: The DAS query tests failed, see the summary page for details.


## RelVals

- [12325.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dfda86/12856/runTheMatrix-results/12325.0_GluGluTo2Jets_14TeV+2021DesignPU+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSim+DigiPU+RecoPU+HARVESTPU)
```
----- Begin Fatal Exception 12-Feb-2021 13:51:24 CET-----------------------
An exception of category 'NoSecondaryFiles' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=MixingModule label='mix'
Exception Message:
RootEmbeddedFileSequence no input files specified for secondary input source.
----- End Fatal Exception -------------------------------------------------
```


## RelVals-INPUT

- 11925.0```DAS Error```
- 12325.0```DAS Error```

qliphy commented:
Similarly as https://github.com/cms-sw/cmssw/pull/32140
but now what missing is: /RelValMinBias_14TeV/CMSSW_11_2_0_pre8-112X_mcRun3_2021_design_v10-v1/GEN-SIM
@cms-sw/pdmv-l2  Do you think you can submit a job to produce that missing file, or the configuration in the workflow should be updated?  Thanks!

chayanit commented:
Hi @qliphy , no we don't need this MinBias. I think there is some misunderstanding with the workflow IDs added in IB test. It shouldn't be the design version. 

@mundim could you double check which workflows should be a regular geometry and not the designed one? Should they be 11725.0 and 12125.0 instead? I'm confused now.

mundim commented:
Hi @chayanit. I was trying to understand this... I'm not sure about your question. The wf ID I took from runTheMatrix.py --what upgrade --showMatrix | grep -i GluGluTo2Jets_M | grep 2021. But it seems the PU wf is trying to get a MinBias from 2011, in which certainly, PPS was not present and I imagine a mixing will fail, right?

chayanit commented:
> Hi @chayanit. I was trying to understand this... I'm not sure about your question. The wf ID I took from runTheMatrix.py --what upgrade --showMatrix | grep -i GluGluTo2Jets_M | grep 2021. But it seems the PU wf is trying to get a MinBias from 2011, in which certainly, PPS was not present and I imagine a mixing will fail, right?

Can you show me what you got after run `runTheMatrix.py --what upgrade --showMatrix | grep -i GluGluTo2Jets_M | grep 2021` only for these 4 workflows 11725.0, 11925.0 and 12125.0, 12325.0?

mundim commented:
Hi. this is the output:

`
11725.0 2021+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSim+Digi+Reco+HARVEST+ALCA 
11925.0 2021PU+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSim+DigiPU+RecoPU+HARVESTPU+Nano 
12125.0 2021Design+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSim+Digi+Reco+HARVEST 
12325.0 2021DesignPU+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSim+DigiPU+RecoPU+HARVESTPU 
`

chayanit commented:
> Hi. this is the output:
> 
> `11725.0 2021+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSim+Digi+Reco+HARVEST+ALCA`
> `11925.0 2021PU+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSim+DigiPU+RecoPU+HARVESTPU+Nano `
> `12125.0 2021Design+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSim+Digi+Reco+HARVEST `
> `12325.0 2021DesignPU+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSim+DigiPU+RecoPU+HARVESTPU`

OK. We should select 11725.0 (noPU) and 11925.0 (PU) as they are standard scenario and not the Design ones. Can you update in relval_2017 file to these two workflows instead? @mundim

chayanit commented on Configuration/PyReleaseValidation/python/relval_2017.py:
```diff
@@ -57,7 +57,7 @@
            11624.911, 11642.911,
            11601.0,11602.0,11603.0,11604.0,11605.0,11606.0,11607.0,11608.0,11609.0,11630.0,11643.0,
-           11650.0,11634.0,11646.0,11640.0,11834.0,11834.99,11846.0,12024.0,
+           11650.0,11634.0,11646.0,11640.0,11834.0,11834.99,11846.0,12024.0,11925.0,12325.0,
```
should be 11725.0, 11925.0 instead

mundim replied:
Done. Thank you.

mundim committed:
Changing the wf number as requested in the PR 32765

mundim commented:
I solved the conflict about relval_2017, but I'm not sure about the other 2 files (I used the latest IB as the base).

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32765/21176

- This PR adds an extra 40KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_2017.py  modified in PR(s): #27983, #31854
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #27983, #31854
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #27983, #32868, #31854

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-checks**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32765/21176/code-checks.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32765/21176/code-checks.patch | patch -p1`
You can also run `scram build code-checks` to apply code checks directly

chayanit commented:
Hello @mundim , @qliphy , @silviodonato any problem with this PR? It seems to conflict with others. How should we fix and move this PR?

silviodonato commented:
After having merged #32969 and #32868 everything should be already merged in master.
@mundim could you check all the code is included in https://github.com/cms-sw/cmssw ?

mundim commented:
> After having merged #32969 and #32868 everything should be already merged in master.
> @mundim could you check all the code is included in https://github.com/cms-sw/cmssw ?

Some how the proposed change in realval_2017.py have been ignored as well as some corrections in the SM parameters in the generator fragment.

qliphy commented:
> > After having merged #32969 and #32868 everything should be already merged in master.
> > @mundim could you check all the code is included in https://github.com/cms-sw/cmssw ?
> 
> Some how the proposed change in realval_2017.py have been ignored as well as some corrections in the SM parameters in the generator fragment.

@mundim As mentioned in https://github.com/cms-sw/cmssw/pull/32969#issuecomment-784150232   you can open a new PR to add all missing fixes.  Thanks!

silviodonato commented:
@mundim
I squashed the commits of this PR and fixed the conflict with master.
Does https://github.com/cms-sw/cmssw/compare/master...silviodonato:PPSRelValGenFragment_reb?expand=1 look ok ?

mundim commented:
> @mundim
> I squashed the commits of this PR and fixed the conflict with master.
> Does https://github.com/cms-sw/cmssw/compare/master...silviodonato:PPSRelValGenFragment_reb?expand=1 look ok ?

It is ok with the gen fragment, but in the relval_2017.py file, lines 33 and 60 got added instead of changed, they are not duplicated.

silviodonato commented:
> the




> > @mundim
> > I squashed the commits of this PR and fixed the conflict with master.
> > Does https://github.com/cms-sw/cmssw/compare/master...silviodonato:PPSRelValGenFragment_reb?expand=1 look ok ?
> 
> It is ok with the gen fragment, but in the relval_2017.py file, lines 33 and 60 got added instead of changed, they are not duplicated.

@mundim Right, thanks a lot. I've fixed it. Will you open the PR or shall I do it?

mundim commented:
> > the
> 
> > > @mundim
> > > I squashed the commits of this PR and fixed the conflict with master.
> > > Does https://github.com/cms-sw/cmssw/compare/master...silviodonato:PPSRelValGenFragment_reb?expand=1 look ok ?
> > 
> > 
> > It is ok with the gen fragment, but in the relval_2017.py file, lines 33 and 60 got added instead of changed, they are not duplicated.
> 
> @mundim Right, thanks a lot. I've fixed it. Will you open the PR or shall I do it?

Done already. thank you very much.

silviodonato commented:
moved to #32983
